app/views/users/show.html.erb

```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if !@user.private? || current_user != @user || current_user.leaders.include?(@user) %>
+<% if policy(@user).show? %>
```

Comment:
Cool! For further improvement, you could dissociate this from the `show?` policy. The issue here is that if someone is private and the user is not following them, they will not even be able to render their profile page, thus they will not be able to request to follow them. Currently, this conditional here is never `triggered` as if the user cannot access the `show` action, this html will not get rendered. Does that make sense?